pubsub/gochannel/fanout.go

```diff
@@ -0,0 +1,67 @@
+package gochannel
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+)
+
+type FanOut struct {
+	pubSub *GoChannel
```

Comment:
maybe we can prefix `pubSub` and `router` with `internal`?

Reply:
666e8e97d7b8637930ea7140a41fe12ed9b6bea4

Reply:
Actually after thinking this through, I think only `pubSub` should be `internal`, as it is created in the constructor?